### PR TITLE
Document AWSConfig CRD

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/8804

TODO:

- [x] Create CRD based on YAML string
- [x] Add unit tests to create/compare CRD and example CR
- [x] Add OpenAPIv3 schema including descriptions for documentation (partly done)

### CODE REVIEW

**Comment by marians** [comment]
Dear reviewers, please, if you know the meaning of any property carrying a `description: TODO`, add a comment or suggestion. Thank you!

**Comment by marians** [comment]
Re-ping @tfussell @tuommaki @tomahawk28 @xh3b4sd

**Comment by xh3b4sd** [comment]
I really love the energy you put into this but I do not want to put my energy in the legacy code related resources. Not worth my time IMO. When this is super important to you please let's get into a call and go through it together. Otherwise I am ignoring this particular PR.

File: docs/crd/provider.giantswarm.io_awsconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+                  workers:
+                    description: |
+                      Configuration of worker nodes.
+                    items:
+                      description: |
+                        Configuration of an individual worker node. Each worker node
+                        is represented by one item.
+                      properties:
+                        dockerVolumeSizeGB:
+                          description: TODO
```
Comment (tfussell) [inline docs/crd/provider.giantswarm.io_awsconfig.yaml]: Size of the worker docker volume in gigabytes.

File: docs/crd/provider.giantswarm.io_awsconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+                        description: |
+                          Elastic Load Balancer configuration.
+                        properties:
+                          idleTimeoutSeconds:
+                            description: |
+                              Seconds to keep an idle connection open.
+                            type: int
+                        type: object
+                      hostedZones:
+                        description: TODO
```
Comment (tfussell) [inline docs/crd/provider.giantswarm.io_awsconfig.yaml]: It's unclear if this is used. I can't figure it out. Maybe get rid of it here?

File: docs/crd/provider.giantswarm.io_awsconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+                        description: |
+                          Configuration of the Elastic Load Balancer for Ingress.
+                        properties:
+                          idleTimeoutSeconds:
+                            description: |
+                              Seconds to keep an idle connection open.
+                            type: int
+                        type: object
+                      hostedZones:
+                        description: TODO
```
Comment (tfussell) [inline docs/crd/provider.giantswarm.io_awsconfig.yaml]: It's unclear if this is used. I can't figure it out. Maybe get rid of it here?

File: docs/crd/provider.giantswarm.io_awsconfig.yaml [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+                    type: object
+                  masters:
+                    description: |
+                      Configuration of the master nodes.
+                    items:
+                      description: |
+                        Configuration for each individual master node.
+                      properties:
+                        dockerVolumeSizeGB:
+                          description: TODO (appears unused)
```
Comment (tfussell) [inline docs/crd/provider.giantswarm.io_awsconfig.yaml]: I don't see this being used.

**Comment by tfussell** [comment]
I tried to look up a few. Unfortunately I never actually used these fields so I'm only guessing (based on the code of course). Someone else might need to do this.

**Comment by tfussell** [comment]
Looks good in general though. It will make things much easier if we have to go back and read old code in the future.

**Comment by marians** [comment]
Since this is legacy, I'm fine with leaving properties without description. Thanks for the effort so far! Now all I need is an approval.

**Review by tfussell — APPROVED** [review]
Seems good for now. We'll improve and/or deprecate as we go.

**Pull request merged**